### PR TITLE
Dhanshri/dev api wrapper

### PR DESCRIPTION
**Pull request by Dhan-shri**
Adding wrapper class

### CODE REVIEW

File: learn/src/main/java/org/navgurukul/learn/ui/learn/LearnFragmentViewModel.kt [hunk 1]
```diff
@@ -217,77 +252,150 @@ class LearnFragmentViewModel(
...
+//                             classes = emptyList()
+//                         )
+//                     }
+//                     if (it.isNotEmpty()) {
+//                         _viewEvents.postValue(LearnFragmentViewEvents.ShowUpcomingBatch(it[0]))
+//                     }
+//                 }
+//                 setState { copy(loading = false) }
+//             } catch (e: Exception) {
                 println(e.message)
```
[the remark this answers is not in the document]
Comment (damanpreet97) [reply learn/src/main/java/org/navgurukul/learn/ui/learn/LearnFragmentViewModel.kt:291]: And also remove the commented code

File: learn/src/main/java/org/navgurukul/learn/ui/learn/LearnFragmentViewModel.kt [hunk 2]
```diff
@@ -217,77 +252,150 @@ class LearnFragmentViewModel(
...
+            val response = learnRepo.getCertificate(pathwayCode)
+            when (response){
+                is Resource.Success -> {
+                    response.data?.let {
+                        _viewEvents.postValue(LearnFragmentViewEvents.GetCertificate(response.data.url, completedPortion, pathwayName))
+                    }
+                    response
+                }
+                is Resource.Error -> {
+                    Log.d("LearnFragmentViewModel", response.message?:"Some error occurred")
```
Comment (damanpreet97) [inline learn/src/main/java/org/navgurukul/learn/ui/learn/LearnFragmentViewModel.kt:381]: Let's show a toast or something here..

File: app/src/main/java/org/merakilearn/ui/profile/ProfileViewModel.kt [hunk 3]
```diff
@@ -262,16 +264,24 @@ class ProfileViewModel(
...
+                when (batches){
+                    is Resource.Success -> {
+                        batches.data?.let {
+                            setState {
+                                copy(
+                                    batches = it
+                                )
+                            }
+                            if (it.isNotEmpty()) {
+                                _viewEvents.postValue(ProfileViewEvents.ShowEnrolledBatches(batches.data!!))
```
[the remark this answers is not in the document]
Comment (Dhan-shri) [reply app/src/main/java/org/merakilearn/ui/profile/ProfileViewModel.kt:276]: Done

File: learn/src/main/java/org/navgurukul/learn/ui/learn/LearnFragmentViewModel.kt [hunk 4]
```diff
@@ -180,28 +192,51 @@ class LearnFragmentViewModel(
...
                     }
                 }
-            } catch (e: Exception) {
-                println(e.message)
-                _viewEvents.postValue(LearnFragmentViewEvents.ShowToast("Error in fetching student enrolment status"))
+                is Resource.Error -> {
+                    setState { copy(loading= false) }
+                    Log.e("LearnFragmentViewModel", "checkedStudentEnrolment: ${status.message}")
+                }
+//             setState { copy(loading = true) }
```
[the remark this answers is not in the document]
Comment (Dhan-shri) [reply learn/src/main/java/org/navgurukul/learn/ui/learn/LearnFragmentViewModel.kt:219]: Removed

File: learn/src/main/java/org/navgurukul/learn/ui/learn/LearnFragmentViewModel.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dhan-shri) [reply learn/src/main/java/org/navgurukul/learn/ui/learn/LearnFragmentViewModel.kt:291]: We are not showing toast every places, I added toast wherever needed.

**Review by damanpreet97 — APPROVED** [review]

**Pull request merged**